packages/fxa-payments-server/src/lib/apiClient.ts

```diff
@@ -0,0 +1,159 @@
+import { defaultConfig } from './config';
+
+// TODO: Use a better type here
+interface APIFetchOptions {
+  [propName: string]: any;
+}
+
+type ErrorResponseBody = {
+  code?: string;
+  statusCode?: number;
+  errno?: number;
+  error?: string;
+  message?: string;
+  info?: string;
+};
+
+export class APIError extends Error {
+  body: ErrorResponseBody | null;
+  response: Response | null | undefined;
+  code: string | null;
+  statusCode: number | null;
+  errno: number | null;
+  error: string | null;
+
+  constructor(
+    body?: ErrorResponseBody,
+    response?: Response,
+    code?: string,
+    errno?: number,
+    error?: string,
+    statusCode?: number,
+    ...params: Array<any>
+  ) {
+    super(...params);
+    this.response = response;
+    this.body = body || null;
+    this.code = code || null;
+    this.statusCode = statusCode || null;
+    this.errno = errno || null;
+    this.error = error || null;
+
+    if (this.body) {
+      const { code, errno, error, message, statusCode } = this.body;
+      Object.assign(this, { code, errno, error, message, statusCode });
+    }
+  }
+}
+
+let accessToken = '';
+let config = defaultConfig();
+
+export function updateAPIClientConfig(configFromMeta: any) {
```

Comment:
I think this is looking a lot better. Main two things are a) would be nice to cover this new extracted module in a unit test and b) I wonder if it might make sense to get even more specific with the config here - i.e. maybe have one function that accepts an accessToken and just config.servers since that's all we use here.

Kind of splitting hairs with the config thing, so I wouldn't prioritize or block on that